src/main/scala/org/broadinstitute/dsde/firecloud/service/MethodsService.scala

```diff
@@ -1,21 +1,108 @@
 package org.broadinstitute.dsde.firecloud.service
 
-import akka.actor.Actor
-import org.broadinstitute.dsde.firecloud.FireCloudConfig
-import org.slf4j.LoggerFactory
+import org.broadinstitute.dsde.firecloud.model.MethodRepository._
 import spray.routing.{HttpService, Route}
+import akka.actor.{Actor, Props}
+import org.broadinstitute.dsde.firecloud.core.{AgoraPermissionActor, AgoraPermissionHandler}
+import org.broadinstitute.dsde.firecloud.model.ModelJsonProtocol._
+import org.broadinstitute.dsde.firecloud.{FireCloudConfig}
+import org.slf4j.{ LoggerFactory}
+import spray.http.StatusCodes._
+import spray.httpx.SprayJsonSupport._
+import spray.json.DefaultJsonProtocol._
 
 class MethodsServiceActor extends Actor with MethodsService {
   def actorRefFactory = context
   def receive = runRoute(routes)
 }
 
-trait MethodsService extends HttpService with FireCloudDirectives {
+
+
+trait MethodsService extends HttpService with PerRequestCreator with FireCloudDirectives {
 
   lazy val log = LoggerFactory.getLogger(getClass)
 
-  val routes: Route =
-    passthroughAllPaths("methods", FireCloudConfig.Agora.methodsBaseUrl) ~
-    passthroughAllPaths("configurations", FireCloudConfig.Agora.configurationsBaseUrl)
+  // Agora routes that require special handling
+
+  val methodsACLOverrideRoute: Route =
+    path("methods" / Segment / Segment / IntNumber / "permissions") { (namespace, name, snapshotId) =>
+
+      // generate the Agora url we'll call
+      val url = FireCloudConfig.Agora.methodsPermissionsFromMethod(namespace, name, snapshotId)
+
+      delete {
+        //this functionality is achieve via "NO ACCESS" in the Post
+        requestContext => requestContext.complete(BadRequest)
```

Comment:
Re "These should be MethodNotAllowed, not BadRequest. ....." last I heard the CLI hadn't gotten started on this, but I can check in with Brad about it and see if he can offer any status update